routes/auth-mentor.js

```
@@ -0,0 +1,341 @@
+    });
+    return;
+  }
+
+  User.findOne({
+    username
+  })
+    .then(user => {
+      if (user !== null) {
+        res.render('auth/signup-mentor', {
+          message: 'The username already exists'
+        });
+        return;
+      }
+
+      const salt = bcrypt.genSaltSync(bcryptSalt);
+      const hashPass = bcrypt.hashSync(password, salt);
+
+      const newUser = new User({
+        username,
+        password: hashPass,
+        role: "Mentor",
+        name,
+        surname
+
+      });
+
+      return newUser.save();
+    })
+    .then(() => {
+      res.redirect('/login');
+    })
+    .catch(error => {
+      res.render('auth/signup-mentor', {
+        message: 'Something went wrong'
+      });
+    });
+});
+
+
+
+///////////////////////LOG IN 
+
+//////// SHOWS YOU THE LOGIN PAGE FORM
+router.get('/login', (req, res, next) => {
+  res.render('auth/login');
+});
+
+
+// router.get('/login', (req, res, next) => {
+//   res.render('auth/login', { message: req.flash('error') });
+// });
+
+///////// THESE ARE THE PRIVATE ROUTES BELOW
+//////// SAME SHOULD BE SET UP FOR MENTEES
+
+router.get('/mentor-space', ensureLogin.ensureLoggedIn(), (req, res) => {
+  User.find({
+    role: "Mentee"
+  })
+    .then(mentees => {
+      console.log(mentees)
+      res.render('spaces/mentor-space', {
+        user: req.user,
+        mentees: mentees.slice(0, 5)
+      })
+    })
+})
+
+
+router.get('/mentor-edit', ensureLogin.ensureLoggedIn(), (req, res) => {
+
+
+
+  res.render('spaces/mentor-edit', {
+
+    user: req.user,
+
+  });
+})
+
+
+
+
+router.get('/common-space', ensureLogin.ensureLoggedIn(), (req, res) => {
+
+  User.find({ role: 'Mentor' }, ' name surname imgPath _id blog')
+
+    .then(mentorBlog => {
+
+      console.log('blog posts' + mentorBlog)
+      res.render('spaces/common-space', {
+        user: req.user,
+
+        mentorBlog: mentorBlog
+      })// handle user
+    })
+    .catch(error => {
```

better remove `catch` if it does nothing, because then it will just silently fail without you ever knowing